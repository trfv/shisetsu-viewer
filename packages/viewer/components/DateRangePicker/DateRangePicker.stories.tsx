import type { Meta, StoryObj } from "@storybook/react";
import { useState } from "react";
import { DateRangePicker } from "./DateRangePicker";

export default {
  component: DateRangePicker,
} as Meta<typeof DateRangePicker>;

export const Default: StoryObj<typeof DateRangePicker> = {
  args: {
    label: "label",
    startDateProps: {
      value: null,
      onChange: () => null,
      minDate: new Date(2021, 0, 1),
      maxDate: new Date(2022, 11, 31),
    },
    endDateProps: {
      value: null,
      onChange: () => null,
      minDate: new Date(2021, 0, 1),
      maxDate: new Date(2022, 11, 31),
    },
  },
  argTypes: {
    startDateProps: {
      control: false,
    },
    endDateProps: {
      control: false,
    },
  },
  render: (args) => {
    const [startDate, setStartDate] = useState<Date | null>(new Date(2021, 0, 1));
    const [endDate, setEndDate] = useState<Date | null>(new Date(2021, 1, 1));
    const onChangeStartDate = (date: Date | null) => setStartDate(date);
    const onChangeEndDate = (date: Date | null) => setEndDate(date);
    return (
      <DateRangePicker
        {...args}
        endDateProps={{ ...args.endDateProps, value: endDate, onChange: onChangeEndDate }}
        startDateProps={{ ...args.startDateProps, value: startDate, onChange: onChangeStartDate }}
      />
    );
  },
};
