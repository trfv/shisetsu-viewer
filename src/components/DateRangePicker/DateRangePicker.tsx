import React, { ComponentProps, FC } from "react";
import { BaseBox, LargeBox } from "../Box";
import { DatePicker } from "../DatePicker";
import { SmallLabel } from "../Label";
import { Spacer } from "../Spacer";

type DatePickerProps = ComponentProps<typeof DatePicker>;

type Props = {
  label: string;
  startDateProps: DatePickerProps;
  endDateProps: DatePickerProps;
};

export const DateRangePicker: FC<Props> = ({ label, startDateProps, endDateProps }) => {
  return (
    <LargeBox display="flex" flexDirection="column">
      <SmallLabel label={label} />
      <Spacer size={4} axis="vertical" />
      <LargeBox display="flex" alignItems="center">
        <DatePicker {...startDateProps} />
        <BaseBox component="span" mr="16px">
          ã€œ
        </BaseBox>
        <DatePicker {...endDateProps} />
      </LargeBox>
    </LargeBox>
  );
};

export default DateRangePicker;
