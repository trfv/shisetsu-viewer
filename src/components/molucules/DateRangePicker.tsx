import Box from "@material-ui/core/Box";
import React, { FC } from "react";
import DatePicker, { DatePickerProps } from "../atoms/DatePicker";
import FormLabel from "../atoms/FormLabel";

type DateRangePickerProps = {
  label: string;
  startDateProps: DatePickerProps;
  endDateProps: DatePickerProps;
};

const DateRangePicker: FC<DateRangePickerProps> = ({
  label,
  startDateProps,
  endDateProps,
}: DateRangePickerProps) => {
  return (
    <>
      <FormLabel labelText={label} />
      <Box display="flex" alignItems="center">
        <DatePicker {...startDateProps} />
        <Box component="span" mx="4px">
          ã€œ
        </Box>
        <DatePicker {...endDateProps} />
      </Box>
    </>
  );
};

export default DateRangePicker;
