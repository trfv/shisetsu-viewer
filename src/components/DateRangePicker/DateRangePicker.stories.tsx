import type { Meta, Story } from "@storybook/react";
import type { ComponentProps } from "react";
import { DateRangePicker } from "./DateRangePicker";

export default {
  title: "DateRangePicker",
  component: DateRangePicker,
} as Meta;

export const Basic: Story<ComponentProps<typeof DateRangePicker>> = (args) => (
  <DateRangePicker {...args} />
);

Basic.args = {
  label: "label",
  startDateProps: {
    value: new Date(2021, 0, 1),
    onChange: () => null,
  },
  endDateProps: {
    value: new Date(2021, 1, 1),
    onChange: () => null,
  },
};
