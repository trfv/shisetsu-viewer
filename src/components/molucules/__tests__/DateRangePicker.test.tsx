import "@testing-library/jest-dom/extend-expect";
import { fireEvent, render } from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";
import DateRangePicker from "../DateRangePicker";

it("snapshot", () => {
  const props = {
    label: "dateRangePicker",
    startDateProps: {
      value: new Date("2020-06-13"),
      onChange: jest.fn(),
    },
    endDateProps: {
      value: new Date("2020-06-20"),
      onChange: jest.fn(),
    },
  };
  const component = renderer.create(<DateRangePicker {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("interaction", () => {
  const props = {
    label: "dateRangePicker",
    startDateProps: {
      value: new Date("2020-06-13"),
      onChange: jest.fn(),
    },
    endDateProps: {
      value: new Date("2020-06-20"),
      onChange: jest.fn(),
    },
  };
  const dateRangePicker = render(<DateRangePicker {...props} />);
  expect(dateRangePicker.baseElement).toBeTruthy();
  fireEvent.click(dateRangePicker.baseElement);
});
