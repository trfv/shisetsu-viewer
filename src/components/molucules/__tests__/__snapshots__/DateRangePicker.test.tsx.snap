// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-1"
  >
    dateRangePicker
  </div>,
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    <div
      className="MuiFormControl-root MuiTextField-root"
      format="yyyy/MM/dd"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={true}
          required={false}
          type="text"
          value="2020/06/13"
        />
      </div>
      
    </div>
    <span
      className="MuiBox-root MuiBox-root-3"
    >
      ã€œ
    </span>
    <div
      className="MuiFormControl-root MuiTextField-root"
      format="yyyy/MM/dd"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={true}
          required={false}
          type="text"
          value="2020/06/20"
        />
      </div>
      
    </div>
  </div>,
]
`;
